easy_my_coop_taxshelter_report/models/tax_shelter_declaration.py

```diff
@@ -322,6 +322,11 @@ def send_certificates(self):
                 certificate.state = "sent"
             else:
                 certificate.state = "no_eligible"
+            # pylint: disable=invalid-commit
+            # fixme while you're here, please fix the query
+            #  to pass pylint invalid-commit
+            #  Use of cr.commit() directly is VERY BAD
+            #  More info https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#never-commit-the-transaction  # noqa
             self.env.cr.commit()
```

Comment:
"Unless: You have created your own database cursor explicitly! And the situations where you need to do that are exceptional! "

justement c'est le but. c'est n'est pas executé par du rpc-client mais via un cron. En sachant que l'on ne veut pas faire de roll back de toute la transaction parce que justement des mails sont envoyés. Et on ne peut pas rollbacker des emails envoyés ici c'est un rollback qui rendre le processus métier inconsistant sachant que chaque ligne à son propre état et est indépendante du statut de la déclaration tax shelter dont elle dépend

Reply:
Mais ici, tu n'as pas créé le curseur explicitement :/

Par contre, t'as raison du coup d'utiliser `commit()`, j'intègre ton commentaire 👍